test/functional/services/embedding.ts

```diff
@@ -17,20 +17,23 @@
  * under the License.
  */
 
-export function EmbeddingProvider({ getService, getPageObjects }) {
+import { FtrProviderContext } from '../ftr_provider_context';
+
+export function EmbeddingProvider({ getService, getPageObjects }: FtrProviderContext) {
```

Comment:
@wayneseymour I was not able to make it work probably because `Embedding` class is been defined inside this function and not accessible yet.

I think we can achieve it by splitting class definition and export function into 2 files, but IMHO it does bring that much value.

I checked other services, only few services have return type on export function and it is where class been imported:

```
import { EsArchiver } from '../../../src/es_archiver';
// @ts-ignore not TS yet
import * as KibanaServer from './kibana_server';

export function EsArchiverProvider({ getService, hasService }: FtrProviderContext): EsArchiver {
```

Reply:
I follow ya, and I think creating 2 files is not desirable.
Here's a question though, what is the true purpose of ts-ifying these files?
If it is to enable "intellisense" in the IDE, I would say typing these returns sounds like a good idea (_in my head anyway_)

Your thoughts?

Reply:
Personally I see several points:
- consistency:  make sure services are used the same way across different tests (kibana/test still have mostly .js files, xpack/test is opposite)
- easy to read and re-use (some services are based on other services)
- way easier to write new .ts tests with "intellisense"

I don't see a problem with not having explicitly set export function return type, VS Code still shows that it is an instance of `Inspector` class:
<img width="1010" alt="hybrid_visualization ts — kibana 2019-09-11 00-04-04" src="https://user-images.githubusercontent.com/10977896/64653863-e1e1de80-d427-11e9-89e5-6c8ba87816b0.png">

`esArchiver` has a return type of `EsArchiver`, but nothing really different - if I got it right
<img width="884" alt="hybrid_visualization ts — kibana 2019-09-11 00-08-40" src="https://user-images.githubusercontent.com/10977896/64654087-6b91ac00-d428-11e9-91b2-1eb2853275c8.png">

Reply:
LGTM